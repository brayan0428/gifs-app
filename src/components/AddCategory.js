import React, { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ addCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length === 0) {
      return;
    }
    addCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};
