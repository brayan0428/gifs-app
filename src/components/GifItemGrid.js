import React from "react";
import PropTypes from "prop-types";

export const GifItemGrid = ({ img, title }) => (
  <div className="GifItemCard">
    <img className="GifItemCard__img" src={img} alt={title} />
    <div className="GifItemCard__body">
      <h4>{title}</h4>
    </div>
  </div>
);

GifItemGrid.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
