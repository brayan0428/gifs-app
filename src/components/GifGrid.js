import React from "react";
import { GifItemGrid } from "./GifItemGrid";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h2>{category}</h2>
      {loading && <h4>Loading...</h4>}
      <div className="GifGrid">
        {gifs.map((gif) => (
          <GifItemGrid key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
