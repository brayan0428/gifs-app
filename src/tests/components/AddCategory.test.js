import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe("<AddCategory />", () => {
  const addCategory = jest.fn();
  let wrapper = shallow(<AddCategory addCategory={addCategory} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory addCategory={addCategory} />);
  });

  test("Validar componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Cambio en el input", () => {
    const input = wrapper.find("input");
    const value = "Hola";
    input.simulate("change", { target: { value } });
  });

  test("No debe llamar la función addCategory", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });
    expect(addCategory).not.toHaveBeenCalled();
  });

  test("Se debe llamar la función addCategory", () => {
    const value = "Hola";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(addCategory).toHaveBeenCalled();
  });
});
