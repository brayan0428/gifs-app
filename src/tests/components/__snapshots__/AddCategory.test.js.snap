// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> Validar componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar gifs..."
    type="text"
    value=""
  />
</form>
`;
